starlite/routes.py

```diff
@@ -0,0 +1,309 @@
+
+        for param in param_match_regex.findall(path):
+            if ":" not in param:
+                raise ImproperlyConfiguredException(
+                    "Path parameters should be declared with a type using the following pattern: '{parameter_name:type}', e.g. '/my-path/{my_param:int}'"
+                )
+            param_name, param_type = (p.strip() for p in param.split(":"))
+            path_format = path_format.replace(param, param_name)
+            path_parameters.append({"name": param_name, "type": param_type_map[param_type], "full": param})
+        return path, path_format, path_parameters
+
+    def create_handler_kwargs_model(self, route_handler: BaseRouteHandler) -> KwargsModel:
+        """
+        Method to create a KwargsModel for a given route handler
+        """
+        dependencies = route_handler.resolve_dependencies()
+        signature_model = get_signature_model(route_handler)
+        path_parameters = {p["name"] for p in self.path_parameters}
+        return KwargsModel.create_for_signature_model(
+            signature_model=signature_model, dependencies=dependencies, path_parameters=path_parameters
+        )
+
+
+class HTTPRoute(BaseRoute):
+    __slots__ = (
+        "route_handler_map",
+        "route_handlers"
+        # the rest of __slots__ are defined in BaseRoute and should not be duplicated
+        # see: https://stackoverflow.com/questions/472000/usage-of-slots
+    )
+
+    @validate_arguments(config={"arbitrary_types_allowed": True})
+    def __init__(
+        self,
+        *,
+        path: str,
+        route_handlers: List[HTTPRouteHandler],
+    ):
+        self.route_handlers = route_handlers
+        self.route_handler_map: Dict[Method, Tuple[HTTPRouteHandler, KwargsModel]] = {}
+        super().__init__(
+            methods=list(chain.from_iterable([route_handler.http_methods for route_handler in route_handlers])),
+            path=path,
+            scope_type=ScopeType.HTTP,
+            handler_names=[get_name(cast(AnyCallable, route_handler.fn)) for route_handler in route_handlers],
+        )
+
+    async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
+        """
+        ASGI app that creates a Request from the passed in args, and then awaits a Response
+        """
+        if scope["method"] not in self.methods:
+            raise MethodNotAllowedException()
+        request: Request[Any, Any] = Request(scope=scope, receive=receive, send=send)
+        route_handler, parameter_model = self.route_handler_map[request.method]
+        if route_handler.resolve_guards():
+            await route_handler.authorize_connection(connection=request)
+
+        caching_enabled = request.method == "GET" and route_handler.cache
+        response: Optional[Union[Response, StarletteResponse]] = None
+        if caching_enabled:
+            response = self.get_cached_response(request=request)
```

Comment:
We should also consider supporting async cache backends.

Reply:
done